### ISSUE TITLE
[Windows] Make internode pod communication work in the Windows environment

### ISSUE THREAD

**Issue by caroline-suse-rancher** [k3s-io/k3s#9295]
As a follow-on from the initial work done in this [issue](https://github.com/k3s-io/k3s/issues/7258), we've identified that internode pod communication may not be working on k3s in Windows OS. Please investigate this and provide a fix.

**Comment by ValeriiVozniuk** [comment]
Hi, could this be the reason that pod on Windows is unable to reach CoreDNS, and resolve both internal and external DNS names?

**Comment by ValeriiVozniuk** [comment]
I see that Linux nodes are annotated
flannel.alpha.coreos.com/backend-data: {"VNI":**1**,"VtepMAC":"aa:f6:23:e5:46:7a"}
And Windows is
flannel.alpha.coreos.com/backend-data: {"VNI":**4096**,"VtepMAC":"00:15:5d:d5:db:a5"}
But I don't see a way to change VNI in k3s
https://docs.k3s.io/networking/basic-network-options

**Comment by brandond** [comment]
Yeah, I think the VNI mismatch is currently a blocker for mixed linux/windows nodes. I think Windows only support 4096?

cc @manuelbuil

**Comment by ValeriiVozniuk** [comment]
> Yeah, I think the VNI mismatch is currently a blocker for mixed linux/windows nodes. I think Windows only support 4096?

That's right, and it also wants name "vxlan0" set for flannel (see https://github.com/microsoft/SDN/tree/master/Kubernetes/flannel/overlay). I've ended up writing "override" configs to /var/lib/rancher/k3s/agent/etc/flannel and /var/lib/rancher/k3s/agent/etc/cni/net.d/ folders, and set them via --flannel-conf/--flannel-cni-conf parameters.
Also, network policies must be disabled on Linux nodes.

**Comment by brandond** [comment]
Ref: https://learn.microsoft.com/en-us/windows-server/networking/sdn/technologies/hyper-v-network-virtualization/hyperv-network-virtualization-technical-details-windows-server 

> Each virtual subnet belongs to a single virtual network (RDID), and it is assigned a unique Virtual Subnet ID (VSID) using either the TNI or VNI key in the encapsulated packet header. The VSID must be unique within the datacenter and is in the range 4096 to 2^24-2.

I guess everything below 4096 is assumed to be a traditional 802.1q VLAN tag?
